app/utils/calculate-burn-height.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { calculateUntilBurnHeightBlockFromCycles } from './calculate-burn-height';
+
+describe(calculateUntilBurnHeightBlockFromCycles.name, () => {
+  const genesisBurnBlockHeight = 666050;
+  test('it calculates correctly with one cycle', () => {
+    const result = calculateUntilBurnHeightBlockFromCycles({
+      genesisBurnBlockHeight,
+      cycles: 1,
+      currentCycleId: 1,
+      rewardCycleLength: 50,
+    });
+    expect(result).toEqual(666151);
+  });
+  test('it calculates correctly with two cycles, one block before cycle', () => {
+    const result = calculateUntilBurnHeightBlockFromCycles({
+      genesisBurnBlockHeight,
+      cycles: 2,
+      currentCycleId: 1,
+      rewardCycleLength: 50,
+    });
+    expect(result).toEqual(666201);
+  });
+  test('it calculates correctly with three cycles using mainnet length, one block into current cycle', () => {
+    const result = calculateUntilBurnHeightBlockFromCycles({
+      genesisBurnBlockHeight,
+      cycles: 3,
+      currentCycleId: 1,
+      rewardCycleLength: 2100,
+    });
+    expect(result).toEqual(674451);
+  });
+  test('it calculates correctly with twelve cycles', () => {
+    const result = calculateUntilBurnHeightBlockFromCycles({
+      genesisBurnBlockHeight,
+      cycles: 12,
+      currentCycleId: 10,
+      rewardCycleLength: 2100,
+    });
+    expect(result).toEqual(693360);
```

Comment:
This doesn't seem right? My napkin math gives 714,350.

Why would the result ever end in `60`? Shouldn't it always end in `50`, with these options?

Reply:
Expected result was indeed wrong :(